docker/drone/src/main/java/org/arquillian/cube/docker/drone/VncRecorderLifecycleManager.java

```diff
@@ -0,0 +1,124 @@
+package org.arquillian.cube.docker.drone;
+
+import org.arquillian.cube.spi.Cube;
+import org.arquillian.cube.spi.CubeRegistry;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.TestResult;
+import org.jboss.arquillian.test.spi.event.suite.After;
+import org.jboss.arquillian.test.spi.event.suite.Before;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+/**
+ * Vnc Cube is set to manual so we can start and stop before and after each test method.
+ * Since recording is finished when container is stopped and we want to have one video per method, each time image is started and stopped.
+ */
+public class VncRecorderLifecycleManager {
+
+    private static final File DEFAULT_LOCATION_OUTPUT_MAVEN = new File("target");
+    private static final File DEFAULT_LOCATION_OUTPUT_GRADLE = new File("build");
+
+    private static final File MAVEN_REPORT_DIR = new File(DEFAULT_LOCATION_OUTPUT_MAVEN, "surefire-reports");
+    private static final File GRADLE_REPORT_DIR = new File(DEFAULT_LOCATION_OUTPUT_GRADLE, "test-results");
+
+    Cube vnc;
+
+    public void startRecording(@Observes Before beforeTestMethod, CubeDroneConfiguration cubeDroneConfiguration, CubeRegistry cubeRegistry) {
+
+        if (cubeDroneConfiguration.isRecording()) {
+
+            // lazy init
+            initVncCube(cubeRegistry);
+
+            vnc.create();
+            vnc.start();
+
+        }
+
+    }
+
+    private void initVncCube(CubeRegistry cubeRegistry) {
+        if (vnc == null) {
+            Cube vncContainer = cubeRegistry.getCube(SeleniumContainers.VNC_CONTAINER_NAME);
+
+            if (vncContainer == null) {
+                throw new IllegalArgumentException("VNC cube is not present in registry.");
+            }
+
+            this.vnc = vncContainer;
+        }
+    }
+
+    public void stopRecording(@Observes After afterTestMethod, TestResult testResult, CubeDroneConfiguration cubeDroneConfiguration, SeleniumContainers seleniumContainers) {
+
+        if (this.vnc != null) {
+
+            if (cubeDroneConfiguration.isRecordingOnlyFailing() && testResult.getStatus() == TestResult.Status.FAILED) {
```

Comment:
how about extracting this condition to method named
`shouldRecordOnlyOnFailure`